### PR TITLE
[zero]added hybrid adam, removed loss scale in adam

### PR DESCRIPTION
**Pull request by Gy-Lu**
Added hybrid adam. Passed the test.
removed loss scale in adam which may be against using habit.

### CODE REVIEW

**Review by feifeibear — APPROVED** [review]

File: tests/test_optimizer/unittest_hybrid_adam.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        optim.step()
+        torch_optim.step()
+
+        if torch.isnan(p.data).any() or torch.isnan(p_copy.data).any():
+            continue
+        assert torch.allclose(p.data, p_copy.data, 1e-4, 1e-2), \
+                              f"adaw mode {adamw}, device {device}, p_dtype {p_dtype}, g_dtype {g_dtype}"
+
+test_adam()
```
Comment (feifeibear) [inline tests/test_optimizer/unittest_hybrid_adam.py:43]: if __name__ == '__main__':
    test_adam()

File: colossalai/nn/optimizer/cpu_adam.py [hunk 2]
```diff
@@ -110,7 +105,7 @@ def step(self, closure=None):
                     assert state['exp_avg_sq'].device.type == 'cpu', "exp_avg should stay on cpu"
                     self.cpu_adam_op.adam_update(self.opt_id, state['step'], group['lr'], beta1, beta2, group['eps'],
                                                  group['weight_decay'], group['bias_correction'], p.data, p.grad.data,
-                                                 state['exp_avg'], state['exp_avg_sq'], self.loss_scale)
+                                                 state['exp_avg'], state['exp_avg_sq'], -1)
```
Comment (feifeibear) [inline colossalai/nn/optimizer/cpu_adam.py:108]: loss scale why -1

Comment (Gy-Lu) [reply colossalai/nn/optimizer/cpu_adam.py:108]: We think setting loss scale in adam is against the using habit.

File: tests/test_optimizer/unittest_hybrid_adam.py (hunk 1 above)
Comment (Gy-Lu) [reply tests/test_optimizer/unittest_hybrid_adam.py:43]: Thanks. for pytest, I would remove it.

**Review by feifeibear — APPROVED** [review]

**Pull request merged**